src/plus/webviews/focus/focusWebview.ts

```diff
@@ -0,0 +1,296 @@
+import { Disposable } from 'vscode';
+import { Commands, ContextKeys } from '../../../constants';
+import type { Container } from '../../../container';
+import { setContext } from '../../../context';
+import { PlusFeatures } from '../../../features';
+import type { SearchedIssue } from '../../../git/models/issue';
+import { serializeIssue } from '../../../git/models/issue';
+import type { SearchedPullRequest } from '../../../git/models/pullRequest';
+import {
+	PullRequestMergeableState,
+	PullRequestReviewDecision,
+	serializePullRequest,
+} from '../../../git/models/pullRequest';
+import type { GitRemote } from '../../../git/models/remote';
+import type { Repository, RepositoryChangeEvent } from '../../../git/models/repository';
+import { RepositoryChange, RepositoryChangeComparisonMode } from '../../../git/models/repository';
+import type { RichRemoteProvider } from '../../../git/remotes/richRemoteProvider';
+import type { Subscription } from '../../../subscription';
+import { SubscriptionState } from '../../../subscription';
+import { registerCommand } from '../../../system/command';
+import { WebviewBase } from '../../../webviews/webviewBase';
+import type { SubscriptionChangeEvent } from '../../subscription/subscriptionService';
+import { ensurePlusFeaturesEnabled } from '../../subscription/utils';
+import type { State } from './protocol';
+import { DidChangeStateNotificationType, DidChangeSubscriptionNotificationType } from './protocol';
+
+interface RepoWithRichRemote {
```

Comment:
Should this go to the `protocol` file?

Reply:
no, this is a local type just for the webview, this shape is serialized into `RepoWithRichProvider` which is in the protocol.